src/display/touch_manager.js

```diff
@@ -0,0 +1,177 @@
+/* Copyright 2024 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+class TouchManager {
+  #container;
+
+  #isPinching = false;
+
+  #overlayManager;
+
+  #onPinching;
+
+  #onPinchEnd;
+
+  #signal;
+
+  #touchInfo = null;
+
+  #touchManagerAC;
+
+  #touchMoveAC = null;
+
+  static #MIN_TOUCH_DISTANCE_TO_PINCH = 0;
+
+  constructor({ container, overlayManager, onPinching, onPinchEnd, signal }) {
+    this.#container = container;
+    this.#overlayManager = overlayManager;
+    this.#onPinching = onPinching;
+    this.#onPinchEnd = onPinchEnd;
+    this.#touchManagerAC = new AbortController();
+    this.#signal = AbortSignal.any([signal, this.#touchManagerAC.signal]);
+
+    container.addEventListener("touchstart", this.#onTouchStart.bind(this), {
+      passive: false,
+      signal: this.#signal,
+    });
+
+    // The 35 is coming from:
+    //  https://searchfox.org/mozilla-central/source/gfx/layers/apz/src/GestureEventListener.cpp#36
+    //
+    // The properties TouchEvent::screenX/Y are in screen CSS pixels:
+    //  https://developer.mozilla.org/en-US/docs/Web/API/Touch/screenX#examples
+    // MIN_TOUCH_DISTANCE_TO_PINCH is in CSS pixels.
+    TouchManager.#MIN_TOUCH_DISTANCE_TO_PINCH ??=
+      35 / (window.devicePixelRatio || 1);
+  }
```

Comment:
How about we instead make this a static getter, since then it'll be initialized "automatically" when actually needed?
```suggestion
  }

  get MIN_TOUCH_DISTANCE_TO_PINCH() {
    // The 35 is coming from:
    //  https://searchfox.org/mozilla-central/source/gfx/layers/apz/src/GestureEventListener.cpp#36
    //
    // The properties TouchEvent::screenX/Y are in screen CSS pixels:
    //  https://developer.mozilla.org/en-US/docs/Web/API/Touch/screenX#examples
    // MIN_TOUCH_DISTANCE_TO_PINCH is in CSS pixels.
    return shadow(this, "MIN_TOUCH_DISTANCE_TO_PINCH", 35 / (window.devicePixelRatio || 1));
  }
```